[MPSInductor] Add `bessel_[jy][01]` ops

malfet opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #149179
* #149123

By simply calling corresponding special functions

Followup TODO: tweak bessel_y0 to match CPU implementation for `torch.half` dtype

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @chenyang78 @kadeng @muchulee8 @amjames @chauhang @aakhundov

malfet committed:
[MPSInductor] Add `bessel_[jy][01]` ops

By simply calling corresponding special functions

But test only j0 in test_mps_basic

[ghstack-poisoned]

dcci commented on torch/_inductor/codegen/mps.py:
```diff
@@ -527,9 +543,9 @@ def reduction(
-            assert not self.multistage_reduction, (
-                f"Multistage reduction not yet supported for {reduction_type}"
-            )
+            assert (
```
I really don't understand why there are these unrelated changes. Did something change in lintrunner? Can we consider reverting that?

malfet replied:
Sorry, those should not have been there, different linter versions

dcci replied:
No worries, I actually experienced the same locally when adding another op.

dcci commented on test/inductor/test_mps_basic.py:
```diff
@@ -119,6 +119,9 @@ def test_pointwise_spherical_bessel_j0(self):
             torch.special.spherical_bessel_j0, (torch.rand(128, 128),), check_lowp=False
         )
 
+    def test_pointwise_bessel_j0(self):
```
Maybe add tests for the other 3 variants if it's not too complicated?

dcci reviewed (approved):
LG

malfet committed:
Update on "[MPSInductor] Add `bessel_[jy][01]` ops"


By simply calling corresponding special functions

But test only j0 in test_mps_basic

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov

[ghstack-poisoned]

malfet commented:
@pytorchbot merge -f "Lint + MPS are green"
